test/api/resources.js

```diff
@@ -238,26 +239,223 @@ describe("API: resource loading configuration", { skipIfBrowser: true }, () => {
         );
       });
     });
+
+    it("should use the custom resource loader when specified", { slow: 500 }, () => {
+      class CustomResource extends ResourceLoader {
+        fetch(url, options) {
+          this.called = true;
+
+          return super.fetch(url, options);
+        }
+      }
+      const sourceString = `Hello`;
+      const url = resourceServer(
+        { "Content-Type": "text/html", "Content-Length": sourceString.length },
+        sourceString
+      );
+      const dom = new JSDOM(`<frameset></frameset>`, { resources: "usable", resourceLoader: new CustomResource() });
+
+      const element = dom.window.document.createElement("frame");
+      setUpLoadingAsserts(element);
+      element.src = url;
+      dom.window.document.body.appendChild(element);
+
+      return assertLoaded(element).then(() => {
+        assert.strictEqual(
+          dom.window.frames[0].document.body.textContent, "Hello",
+          "The frame must have been downloaded"
+        );
+        assert.isTrue(dom.window._resourceLoader.called, "The custom resource should be called");
```

Comment:
Don't probe underscore-prefixed properties, even in tests. Instead, you can use a closure variable at the top of this test, and just set `called = true` instead of `this.called = true`.